source/frontend/src/features/mapSideBar/acquisition/common/GenerateForm/GenerateFormContainer.tsx

```diff
@@ -14,16 +22,155 @@ export interface IGenerateFormContainerProps {
 const GenerateFormContainer: React.FunctionComponent<
   React.PropsWithChildren<IGenerateFormContainerProps>
 > = ({ acquisitionFileId, View }) => {
-  const generateLetter = useGenerateLetter();
+  const [fullRecipientsList, setFullRecipientsList] = useState<LetterRecipientModel[]>([]);
+  const { getPersonConcept, getOrganization } = useApiContacts();
+  const {
+    getAcquisitionFile: { execute: getAcquisitionFile },
+  } = useAcquisitionProvider();
+  const [isGenerateLetterModalOpened, openGenerateLetterModal, closeGenerateLetterModal] =
+    useModalManagement();
+  const {
+    getAcquisitionOwners: {
+      execute: retrieveAcquisitionFileOwners,
+      loading: loadingAcquisitionFileOwners,
+    },
+  } = useAcquisitionProvider();
+  const {
+    getAcquisitionInterestHolders: {
+      execute: fetchInterestHolders,
+      loading: loadingInterestHolders,
+    },
+  } = useInterestHolderRepository();
 
-  const generateH0443 = useGenerateH0443();
+  const fetchAllRecipients = useCallback(async () => {
+    let i = 0;
+    const generateRecipientsList: LetterRecipientModel[] = [];
 
-  const generateH120 = useGenerateH120();
+    const file = await getAcquisitionFile(acquisitionFileId);
+    if (file === undefined) {
+      return;
+    }
+
+    const fileOwnersFetchCall = retrieveAcquisitionFileOwners(acquisitionFileId);
+    const interestHoldersFetchCall = fetchInterestHolders(acquisitionFileId);
+
+    // Add owners and interest holders
+    await Promise.all([fileOwnersFetchCall, interestHoldersFetchCall]).then(
+      ([fileOwners, intHolders]) => {
+        if (fileOwners !== undefined) {
+          const ownersLetterModel = fileOwners?.map(owner => new Api_GenerateOwner(owner)) ?? [];
+          const ownerRecipientsLetterModel =
+            ownersLetterModel.map(x => new LetterRecipientModel(i++, 'OWNR', x)) ?? [];
+          generateRecipientsList.push(...ownerRecipientsLetterModel);
+        }
+
+        if (intHolders !== undefined) {
+          intHolders.map(async holder => {
+            if (holder.personId) {
+              const person = !!holder?.personId
+                ? (await getPersonConcept(holder?.personId))?.data
+                : null;
+
+              if (person) {
+                generateRecipientsList.push(
+                  new LetterRecipientModel(
+                    i++,
```

Comment:
If I'm understanding what this is I'm not a big fan, as this seems like a way of trying to create a unique key for each of these different entities. I think I'd rather have a function that could return a unique key from a LetterRecipientModel that uses the fields of that object to build a unique key. Maybe I'm not understanding something however.

Reply:
updated